llvm/lib/TableGen/TGParser.cpp

```diff
@@ -4359,3 +4390,27 @@ LLVM_DUMP_METHOD void MultiClass::dump() const {
     E.dump();
 }
 #endif
+
+bool TGParser::ParseDump(MultiClass *CurMultiClass, Record *CurRec) {
+  assert(Lex.getCode() == tgtok::Dump && "Unknown tok");
+  Lex.Lex(); // eat the operation
+
+  Init *Message = ParseValue(CurRec);
+  if (!Message)
+    return true;
+
+  std::string Msg;
+  if (auto SI = dyn_cast<StringInit>(Message))
+    Msg = SI->getValue();
+
+  if (!consume(tgtok::semi))
+    return TokError("expected ';'");
+
+  SMLoc Loc = Lex.getLoc();
```

Comment:
This `Loc` is not exact. Should it be the location of `dump` or message?

Reply:
Done - I have checked the lines in the tests.